src/Featurizers/AnalyticalRollingWindowFeaturizer.h

```diff
@@ -0,0 +1,307 @@
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<double>>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    AnalyticalRollingWindowTransformer(std::uint32_t windowSize, AnalyticalRollingWindowCalculation windowCalculation, std::uint32_t horizon, std::uint32_t minWindowCount = 1) ;
+    AnalyticalRollingWindowTransformer(Archive &ar);
+    ~AnalyticalRollingWindowTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(AnalyticalRollingWindowTransformer);
+
+    bool operator==(AnalyticalRollingWindowTransformer const &other) const;
+    bool operator!=(AnalyticalRollingWindowTransformer const &other) const;
+    void save(Archive & ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using OutputType = std::vector<double>;    
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Members
+    // |
+    // ----------------------------------------------------------------------
+    // _windowSizes and _minWindowCount are int64 so that we can have the full positive range of the uint32 it comes
+    // in as but can still support negative numbers.
+    Components::CircularBuffer<InputT>      _buffer;
+    std::int64_t                            _windowSize;
+    std::uint32_t                           _horizon;
+    std::int64_t                            _minWindowCount;
+    AnalyticalRollingWindowCalculation      _windowCalculation;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        OutputType         results(_horizon);
+        
+        _buffer.push(input);
+        size_t bufferSize = _buffer.size();
+
+        for (int i = _horizon; i > 0; i--){
+            // We cast to int32 here because -1 is valid for the algorithm.
+            std::int32_t elementsAvailableToUse(static_cast<std::int32_t>(bufferSize) - i);
+
+            // If we don't have enough elements then output NaN
+            if (elementsAvailableToUse < _minWindowCount) {
+                results[_horizon - i] = std::nan("");

```

Comment:
>std::nan("") [](start = 40, length = 12)

`Traits<InputT>::CreateNullValue()`

Reply:
I can't use `InputT` here as the input type doesn't matter, its always going to double. I have changed it to be `Traits<double>::CreateNullValue()`. I could do a mapping here if needed, though in this case it would always be InputT to double. If you have any other thoughts on that let me know.